pkg/util/aad/aad.go

```diff
@@ -57,6 +58,7 @@ func GetToken(ctx context.Context, log *logrus.Entry, clientID, clientSecret, te
 		claims := jwt.MapClaims{}
 		_, _, err = p.ParseUnverified(authorizer.OAuthToken(), claims)
 		if err != nil {
+			err = errServicePrincipal
```

Comment:
i am not sure if this level of simplification is needed. I personally prefer to return a new cloud error here with more detail from `err`